### PR TITLE
create-content-artifacts for MP V2

### PR DESCRIPTION
**Pull request by Noy-Maimon**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/44218

## Description
As part of the MP v2 effort:
Added a mp flag that determines the MP we are uploading to.
Added a condition that if mp=v2, we do not create the unwanted artifacts (as listed in the issue)

Note: there is no filtering on the pack metadata 'marketplaces' fields since the intention is to supply to `create-content-artifacts` a list of packs as an argument (which is an already existing argument). This list will be generated in the `create-id-set` part, done priorly in the flow.

### CODE REVIEW

File: demisto_sdk/commands/create_artifacts/content_artifacts_creator.py [hunk 1]
```diff
@@ -671,28 +663,39 @@ def dump_pack(artifact_manager: ArtifactsManager, pack: Pack) -> ArtifactsReport
...
+        for dashboard in pack.dashboards:
+            content_items_handler.handle_content_item(dashboard)
+            pack_report += dump_pack_conditionally(artifact_manager, dashboard)
+        for widget in pack.widgets:
+            content_items_handler.handle_content_item(widget)
+            pack_report += dump_pack_conditionally(artifact_manager, widget)
+        for report in pack.reports:
+            content_items_handler.handle_content_item(report)
+            pack_report += dump_pack_conditionally(artifact_manager, report)
+        for layout in pack.layouts:  # TODO: verify layouts == layouts_container
```
Comment (yaakovi) [inline demisto_sdk/commands/create_artifacts/content_artifacts_creator.py:683]: yes, layouts containers are the layouts of 6.0+

File: demisto_sdk/commands/create_artifacts/content_artifacts_creator.py [hunk 2]
```diff
@@ -639,21 +640,12 @@ def dump_pack(artifact_manager: ArtifactsManager, pack: Pack) -> ArtifactsReport
         pack_report += dump_pack_conditionally(artifact_manager, playbook)
     for test_playbook in pack.test_playbooks:
         pack_report += dump_pack_conditionally(artifact_manager, test_playbook)
-    for report in pack.reports:
-        content_items_handler.handle_content_item(report)
-        pack_report += dump_pack_conditionally(artifact_manager, report)
-    for layout in pack.layouts:
-        content_items_handler.handle_content_item(layout)
-        pack_report += dump_pack_conditionally(artifact_manager, layout)
     for pre_process_rule in pack.pre_process_rules:
```
Comment (yaakovi) [inline demisto_sdk/commands/create_artifacts/content_artifacts_creator.py:643]: should also move to `xsoar` only

File: demisto_sdk/commands/create_artifacts/content_artifacts_creator.py [hunk 3]
```diff
@@ -671,28 +663,39 @@ def dump_pack(artifact_manager: ArtifactsManager, pack: Pack) -> ArtifactsReport
...
+        for generic_definition in pack.generic_definitions:
+            content_items_handler.handle_content_item(generic_definition)
+            pack_report += dump_pack_conditionally(artifact_manager, generic_definition)
+        for generic_module in pack.generic_modules:
+            content_items_handler.handle_content_item(generic_module)
+            pack_report += dump_pack_conditionally(artifact_manager, generic_module)
+        for generic_type in pack.generic_types:
+            content_items_handler.handle_content_item(generic_type)
+            pack_report += dump_pack_conditionally(artifact_manager, generic_type)
+        for generic_field in pack.generic_fields:
```
Comment (yaakovi) [inline demisto_sdk/commands/create_artifacts/content_artifacts_creator.py:695]: blame my OCD...
please reorder the loops by content items alphabetically

File: demisto_sdk/__main__.py [hunk 4]
```diff
@@ -549,6 +549,10 @@ def validate(config, **kwargs):
               type=click.Path(exists=True, resolve_path=True), hidden=True)
 @click.option('-rt', '--remove-test-playbooks', is_flag=True,
               help='Should remove test playbooks from content packs or not.', default=True, hidden=True)
+@click.option('-mp', '--marketplace', help='The marketplace the artifacts are created for, that '
+                                           'determines which artifacts are created for each pack.'
```
Comment (yaakovi) [inline demisto_sdk/__main__.py:553]:
```suggestion
                                           'determines which artifacts are created for each pack. '
```

File: demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md [hunk 5]
```diff
@@ -59,4 +59,94 @@ If you are interested in contributing, visit our contribution process [here](htt
 
 Thank you for contributing to Cortex XSOAR.
 
+If you are interested in contributing, visit our contribution process [here](https://xsoar.pan.dev/docs/contributing/contributing).#### This pack was co-authored by:
```
Comment (yaakovi) [inline demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md:62]: revert

**Review by yaakovi — APPROVED** [review]

File: demisto_sdk/commands/common/constants.py [hunk 6]
```diff
@@ -70,6 +70,10 @@
 GENERIC_DEFINITION = 'genericdefinition'
 JOB = 'job'
 
+MARKETPLACE_KEY_PACK_METADATA = 'marketplaces'
+MARKETPLACE_XSIAM_VALUE_PACK_METADATA = 'marketplacev2'
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/constants.py]: already modified

**Review by Bargenish — APPROVED** [review]

**Pull request merged**